src/Form/Control/Lookup.php

```diff
@@ -450,4 +451,9 @@ public function set($value = null, $junk = null)
 
         return parent::set($value, $junk);
     }
+
+    protected function mergeStickyArgsFromChildView(): ?AbstractView
```

Comment:
do we need this?

Reply:
for generating proper callback url, yes.

Reply:
URL of that particial callback (`Lookup->callback`) will always be ok - as this callback is used for API only, I do not think this correct - sticky args will be added from all parents to that callback

Reply:
It will be needed when using lookup inside CRUD edit/add form. 
Otherwise, dropdown is always empty because the modal callback terminate before the lookup callback. This is due to lookup callback not having proper url set. I have test it using another project and Lookup is working fine after this fix. 

You can also see for yourself in: https://ui.agiletoolkit.org/demos/collection/actions.php
In Crud, select Actions/Edit and click the Dropdown for "Parent Folder Id" control. Nothing shows. With this fix, parent folder are properly displayed in dropdown control.

Reply:
> I have test it using another project and Lookup is working fine after this fix.

it always will, the question is is it needed? this method was introduced by me, it has a super power and I would like to use it as less as possible, otherwise we can be where we was

> This is due to lookup callback not having proper url set.

I think this is the issue - you probably want to fix your code or use `CallbackLater`, but not this method

Reply:
CallbackLater will not do any good either. The problem is from generating the proper URL for Callback in order to execute it when it is triggered.

Thus, I do not see any other alternative than using mergeStickyArgsFromChildView(). Besides, there were other cases in a specific project when Callback was needed for a View and that I had to use this method too for generating proper callback URL. 

Should we revert #1397 and keep app sticky argument? The way I see it, any future View using a Callback where proper URL needs to be generated might has to need mergeStickyArgsFromChildView(). It is true for Modal, Loader, Popup, Panel/Content, Wizard and now Lookup...

Reply:
> Should we revert #1397 and keep app sticky argument?

never! This PR is very critical to support true component tree.

If the `mergeStickyArgsFromChildView` method is really needed here, add the most simpletest Behat test for it and let's merge it. This way, we can be sure the functionality is fine and have still a testable option to remove this method in the future.

Reply:
Test added!

Reply:
great! One CI job failed - https://github.com/atk4/ui/pull/1486/checks?check_run_id=1287604565 - probably some issue with detection if load finished or not. Once fixed, you can 🚀 from my side